### PR TITLE
Initial Unit Tests

### PR DESCRIPTION
**Pull request by furkansahin**
## Initial Unit Tests
This commit adds initial unit tests. In Go, I couldn't find a
straightforward way to test main() function. So, I have separated the
whole logic in it into a mainInternal() function and tested that. To be
able to exclude the main() function from coverage, I separated it into a
different file and added a comment to the top so that go-ignore-cov tool
picks it up to exclude.

I have added a workflow file as well so that the tests would run in
every change. We are also enforcing 100% coverage.

## Remove go-releaser run for PRs

### CODE REVIEW

**Review by fdr — APPROVED** [review]
dayum, 100%? that surpasses my expectations for a Go program. No problem in carving out key exemptions (as we do with :nocov: in Ruby), so long as inspection is ~= automated.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - "*"
```
Comment (enescakir) [inline .github/workflows/test.yml]: No need these lines. `pull_request:` should be enough for all branches

**Review by enescakir — APPROVED** [review]

File: metadata-endpoint.go [hunk 2]
```diff
@@ -35,15 +35,15 @@ func fileHandler(filename string) http.HandlerFunc {
...
 	vmInhostName := os.Getenv("VM_INHOST_NAME")
 	if vmInhostName == "" {
-		log.Fatal("Environment variable vm_path is not set")
+		return 1, fmt.Errorf("Environment variable VM_INHOST_NAME is not set")
 	}
 
 	ipv6Address := os.Getenv("IPV6_ADDRESS")
 	if ipv6Address == "" {
-		log.Fatal("Environment variable ipv6_address is not set")
+		return 1, fmt.Errorf("Environment variable ipv6_address is not set")
```
Comment (bsatzger) [inline metadata-endpoint.go]: Nit: 
```suggestion
		return 1, fmt.Errorf("Environment variable IPV6_ADDRESS is not set")
```

**Review by bsatzger — APPROVED** [review]
💯

**Pull request merged**